src/cljs/athens/page.cljs

```diff
@@ -70,14 +70,40 @@
        [:div {:style {:margin-left 20}}
         [render-blocks (:block/uid @node)]]])))
 
+(def enter-keycode 13)
+(def esc-keycode 27)
+
+(defn title-comp [title]
+  (let [s (reagent/atom {:editing false
+                         :current-title title
+                         :new-title title})
+        save! (fn [] (dispatch [:node/rename (:current-title @s) (:new-title @s)])
+                     (swap! s assoc :editing false))
+        cancel! (fn [] (swap! s #(-> %
+                                     (assoc :editing false)
+                                     (assoc :new-title (:current-title @s)))))]
+    (fn [title]
```

Comment:
Small suggestion: I would usually name a function when it extends 1 or 2 lines of code. _Ex:_ `(fn my-fn [title] ...)`. The reason is while debugging the name for non-anonymous functions shows up in the trace. What do you think?

Reply:
@avichalp, thanks for the suggestion! I'm not sure about this one. This actually represents a reagent render function, which is a well known pattern and I've never seen (as far as I can remember) a name used here. Curious to hear @jeroenvandijk's opinion about this.

Reply:
I'm not sure, I can imagine it could give some extra information during debugging. I have done things like this in deeply nested jvm code, for Clojurescript I haven't been there yet. Right now it feel arbitrary for me, so neutral voice here.

@avichalp does it matter if source maps are enabled or not? Or do you do this for any component fn? Or do you mean any anonymous fn?

Let's wait for some bugs and decide on this later. I'm working on integrating an error tracker #78 Might give some use cases for this proposed method of @avichalp.

Reply:
I think you guys are right. I had JVM in mind, actually 😅. It might be overkill—also error messages in CLJS are not that horrifying like JVM stack traces.

Reply:
Thanks @jelmerderonde for your PR 🙏